src/main/java/seedu/address/logic/commands/AddCommand.java

```diff
@@ -2,66 +2,153 @@
+            + PREFIX_LEVEL + "Primary3 "
+            + PREFIX_NEXTOFKIN + "Mom Doe "
+            + PREFIX_TAG + "badBoy "
             + PREFIX_TAG + "owesMoney";
 
-    public static final String MESSAGE_SUCCESS = "New person added: %1$s";
-    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book";
+    public static final String MESSAGE_SUCCESS = "New entity added: %1$s";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This student or tutor already exists in the database";
+    public static final String MESSAGE_DUPLICATE_CLASS = "This class already exists in the database";
 
-    private final Person toAdd;
+    private Person personToAdd;
+    private TuitionClass classToAdd;
 
     /**
      * Creates an AddCommand to add the specified {@code Person}
      */
-    public AddCommand(Person person) {
+    private AddCommand() {
+        this.personToAdd = null;
+        this.classToAdd = null;
+    }
+
+    public static AddCommand of(Person person) {
         requireNonNull(person);
-        toAdd = person;
+        AddCommand addCommand = new AddCommand();
+        addCommand.personToAdd = person;
+        return addCommand;
+    }
+
+    public static AddCommand of(TuitionClass tuitionClass) {
+        requireNonNull(tuitionClass);
+        AddCommand addCommand = new AddCommand();
+        addCommand.classToAdd = tuitionClass;
+        return addCommand;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        if (model.hasPerson(toAdd)) {
-            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+        if (this.classToAdd == null) {
+            requireNonNull(this.personToAdd);
+            if (model.hasPerson(this.personToAdd)) {
+                throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+            }
+            model.addPerson(this.personToAdd);
+            return new CommandResult(String.format(MESSAGE_SUCCESS, this.personToAdd));
+        } else {
+            requireNonNull(this.classToAdd);
+            if (model.hasTuitionClass(this.classToAdd)) {
+                throw new CommandException(MESSAGE_DUPLICATE_CLASS);
+            }
+            model.addTuitionClass(this.classToAdd);
+            return new CommandResult(String.format(MESSAGE_SUCCESS, this.classToAdd));
         }
-
-        model.addPerson(toAdd);
-        return new CommandResult(String.format(MESSAGE_SUCCESS, toAdd));
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof AddCommand // instanceof handles nulls
-                && toAdd.equals(((AddCommand) other).toAdd));
+                && (Objects.equals(this.personToAdd, ((AddCommand) other).personToAdd))
+                && (Objects.equals(this.classToAdd, ((AddCommand) other).classToAdd)));
```

Comment:
Maybe you would want to change to use equals() method in Person and TuitionClass instead? Because from google: Objects.equals() implements the most discriminating possible equivalence relation on objects; that is, for any non-null reference values x and y , this method returns true if and only if x and y refer to the same object ( x == y has the value true ).

Reply:
https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#equals-java.lang.Object-java.lang.Object-

Reply:
icic soz for the confusion